test/unittests/parser_test.cpp

```diff
@@ -440,39 +440,44 @@ TEST(parser, function_section_empty)
 
 TEST(parser, function_section_with_single_function)
 {
+    const auto type_section = make_vec({make_functype({}, {})});
     const auto function_section = make_vec({"00"_bytes});
     const auto code_section = make_vec({"02000b"_bytes});
-    const auto bin =
-        bytes{wasm_prefix} + make_section(3, function_section) + make_section(10, code_section);
+    const auto bin = bytes{wasm_prefix} + make_section(1, type_section) +
+                     make_section(3, function_section) + make_section(10, code_section);
     const auto module = parse(bin);
     ASSERT_EQ(module.funcsec.size(), 1);
     EXPECT_EQ(module.funcsec[0], 0);
 }
 
 TEST(parser, function_section_with_multiple_functions)
 {
-    const auto function_section = "04000142ff01"_bytes;
+    const auto type_section = make_vec({make_functype({}, {}), make_functype({}, {}),
```

Comment:
this could have been just 2 types

Reply:
The original test had unique type_idx. I kept them distinct at least.